python/ray/serve/metric/sink.py

```diff
@@ -0,0 +1,139 @@
+from typing import Dict
+from collections import Counter, namedtuple
+
+import ray
+from ray.serve.metric.types import MetricType, MetricMetadata, MetricBatch
+from ray.serve.utils import _get_logger
+
+logger = _get_logger()
+
+
+def make_metric_namedtuple(metadata: MetricMetadata, record: MetricBatch):
+    fields = ["name", "type"]
+    fields += list(metadata.default_labels.keys())
+    fields += list(record.labels.keys())
+
+    merged_labels = metadata.default_labels.copy()
+    merged_labels.update(record.labels)
+
+    tuple_type = namedtuple(metadata.name, fields)
+    return tuple_type(name=metadata.name, type=metadata.type, **merged_labels)
+
+
+class BaseSink:
+    """A sink is the place to store or forward metrics to external services"""
+
+    def __init__(self):
+        self.metadata = dict()
+        self.recording = []
+        logger.debug("Sink initialized {}".format(type(self)))
+
+    def push_batch(self, metadata: Dict[str, MetricMetadata],
+                   batch: MetricBatch):
+        self.metadata.update(metadata)
+        self.recording.extend(batch)
+        self.compact()
+
+    def compact(self):
```

Comment:
`compact` sounds like a specific implementation of of this method rather than a description of what's to be done. How about `process_metric_batches`?